src/map/ai/helpers/pathfind.cpp

```diff
@@ -470,25 +470,40 @@ bool CPathFind::FindRandomPath(const position_t& start, float maxRadius, uint8 m
 
     auto m_turnLength = xirand::GetRandomNumber((int)maxTurns) + 1;
 
-    position_t startPosition = start;
+    // Seemingly arbitrary value to pass for maxRadius, all values seem to give similar results, likely due to navmesh polygons being too dense?
+    float      maxRadiusForPolyQuery = maxRadius / 10.f;
+    position_t startPosition         = start;
 
-    // find end points for turns
-    for (int i = 0; i < m_turnLength; i++)
+    // find end points for turns, iterate potentially twice as many times to account for erroneous turnPoints
+    for (int i = 0; i < m_turnLength * 2; i++)
     {
-        // look for new point centered around the last point
-        auto status = m_POwner->loc.zone->m_navMesh->findRandomPosition(startPosition, maxRadius);
+        // look for new turnPoint. findRandomPosition doesn't guarantee the new point is within the radius
+        auto status = m_POwner->loc.zone->m_navMesh->findRandomPosition(startPosition, maxRadiusForPolyQuery);
 
         // couldn't find one point so just break out
         if (status.first != 0)
         {
             return false;
         }
 
-        m_turnPoints.push_back(status.second);
-        startPosition = m_turnPoints[i];
+        float distSq = distanceSquared(startPosition, status.second, true);
+        // only add the roam point if it's _actually_ within range of the spawn point...
+        if (distSq < maxRadius * maxRadius)
+        {
+            m_turnPoints.push_back(status.second);
+        }
+        // else
+        // {
+        //     ShowDebug("CPathFind::FindRandomPath (%s - %d) random point too far: sq distance (%f)", m_POwner->GetName(), m_POwner->id, distSq);
+        // }
+        if (m_turnPoints.size() >= m_turnLength)
```

Comment:
win32 CI doesnt like this line -- `auto m_turnLength` might resolve differently there, We probably need to make `m_turnLength` a uint32_t (or appropriate)

Interestingly, `maxTurns`  is defined as uint8 but casted to `int`

you can probably remove the cast there and use `xirand::GetRandomNumber<uint8_t>(maxTurns)` and then `auto` should resolve it the same everywhere

Reply:
Useful to note that STL `.size()` calls always return `std::size_t`, which tends to be equivalent to `unsigned int` (length dependant on system arch)

https://en.cppreference.com/w/c/types/size_t

Reply:
Thanks, good info. I'll resolve this compiler warning tonight.

Reply:
FWIW i couldn't static cast the rand with `uint8_t`, then used `uint32_t` and got this warning:

![image](https://github.com/LandSandBoat/server/assets/131182600/7cd1d01f-1777-4cee-94e7-9ff58d1f49ee)

I found a similar situation in `mobentity.cpp` 

```
// The underlying function in GetRandomNumber doesn't accept uint8 as <T> so use uint32
// https://stackoverflow.com/questions/31460733/why-arent-stduniform-int-distributionuint8-t-and-stduniform-int-distri
uint8 level = static_cast<uint8>(xirand::GetRandomNumber<uint32>(m_minLevel, m_maxLevel + 1));
```

And coded it similarly

Compiled and confirmed roaming still functions as expected.

Reply:
CI failed again. I'm able to replicate this locally choosing the `x86-Release` build.

compiler debugging is not my strong suit, but i'll poke at this for a little bit

Reply:
Without looking at the code, the comparison is with the size-t from the size() method, not the incoming turn count. I'll write up what you need to change tomorrow

Reply:
seems like auto is not choosing uint8 when i do uint8 + 1, but if i wrap the whole thing in a static_cast it works <.<

Reply:
1 is unintuitively not unsigned, you could skirt this with uint8 + 1U, since numeric literals are signed by default

Reply:
(but if everything is passing CI, don't worry about jiggling things around anymore)